tools/elfdeps.c

```diff
@@ -324,8 +431,11 @@ static int processFile(const char *fn, int dtype)
 	    const char *bn = strrchr(fn, '/');
 	    ei->soname = rstrdup(bn ? bn + 1 : fn);
 	}
-	if (ei->soname)
+	if (ei->soname) {
 	    addDep(&ei->provides, ei->soname, NULL, ei->marker);
+	    if (multiarch_deps)
+		addDep(&ei->provides, ei->soname, NULL, ei->archmarker);
+	}
```

Comment:
Oh and ditto here too, shouldn't the non-archmarker addDep() be behind an else?

If the answer is "no" for both these cases I think it'd be good to explain why that is (in the commit message or code comment)

Reply:
I'll add to the commit message about the backwards compatibility concern.